fvm/src/storagemarket_actor.rs

```diff
@@ -0,0 +1,110 @@
+use std::str::FromStr;
+
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::tuple::*;
+use fvm_ipld_encoding::{BytesSer, Cbor};
+use fvm_shared::clock::{ChainEpoch};
+use fvm_shared::deal::DealID;
+use fvm_shared::econ::TokenAmount;
+use fvm_ipld_amt::Amt;
+use fvm_ipld_hamt::Hamt;
+use fvm_shared::piece::PaddedPieceSize;
+use libipld_core::ipld::Ipld;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use fvm_shared::address::Address;
+
+pub type AllocationID = u64;
+
+/// Market actor state
+#[derive(Clone, Default, Serialize_tuple, Deserialize_tuple, Debug)]
```

Comment:
Ouch, we absolutely don't want to have this here. And we're getting to a point where it's best for the integration test framework to stay light _without actor_ assumptions, and to have a separate utility module (in a separate repo) that composes actors with the integration test framework to "prepare" a tester.

I would be less concerned if we moved this type to _inside_ the integration test framework module, only temporarily until we figure out the final layout. WDYT, @Stebalien?

Reply:
It is also duplicated code from the builtin actors.

Reply:
We don't need this. Use the bundle and call the constructor on the market actor to construct the state.

Reply:
That might not work. However I'm quite familiar with this problem because I'm actively working on tooling for profiling built-in actors. I'm not at reveal stage yet (e.g. no docs) but I'm prototyping a new thing at https://github.com/anorth/fvm-workbench

It specifically solves the problem of getting all the built-in actors into a VM for testing (and my next step is to connect it to existing builtin actors scenario tests)

Reply:
Ah... because we assert that the immediate caller is the system actor when it'll be the init actor.

Reply:
So, looking at the singletons, I'm pretty sure we can construct them, we just need to call the constructor from the "system" actor.

Reply:
Yeah, it'll work. Basically:

1. Import the actors bundle.
2. Construct each singleton by calling the "Constructor" method on the singleton (method 1).

That will let us avoid having the actual _state_ here.

Reply:
Yeah, it'll work. Basically:

1. Import the actors bundle.
2. Construct each singleton by calling the "Constructor" method on the singleton (method 1).

That will let us avoid having the actual _state_ here.